### DISCUSSION TITLE
About the functionality of `scale_model_input` in `EulerDiscreteScheduler`

### DISCUSSION THREAD

**Discussion by luocfprime** [huggingface/diffusers#7273] [Q&A]
May I ask what exactly does scale_model_input do in the EulerDiscreteScheduler (and other schedulers in general)?

I've checked the original EDM paper and k-diffusion implementation, the scaling coefficient 
 is 1 in the paper. In [k-diffusion sample_euler implementation](https://github.com/crowsonkb/k-diffusion/blob/6ab5146d4a5ef63901326489f31f1d8e7dd36b48/k_diffusion/sampling.py#L117-L135), the model input does not require extra scaling.

```python
    for i in trange(len(sigmas) - 1, disable=disable):
        gamma = min(s_churn / (len(sigmas) - 1), 2 ** 0.5 - 1) if s_tmin <= sigmas[i] <= s_tmax else 0.
        eps = torch.randn_like(x) * s_noise
        sigma_hat = sigmas[i] * (gamma + 1)
        if gamma > 0:
            x = x + eps * (sigma_hat ** 2 - sigmas[i] ** 2) ** 0.5
        denoised = model(x, sigma_hat * s_in, **extra_args) # no extra scaling for x
```

But in diffusers implementation, the input latent is scaled by `1 / ((sigma**2 + 1) ** 0.5)`, may I ask the rationale behind this? What exactly does `scale_model_input` do?

Also in EDM paper, the add-noise can be generally parameterized as $x_t = s(t) x_0 + s(t) \sigma(t) \epsilon, \quad \epsilon \sim \mathcal{N}(0, I)$. I wonder how are `scheduler.scale_model_input` and `scheduler.add_noise` related to this parameterization?

Thanks!

**Comment by sayakpaul** [comment]
Cc: @yiyixuxu 

For the EDM variant of the Euler scheduler, refer to https://github.com/huggingface/diffusers/blob/main/src/diffusers/schedulers/scheduling_edm_euler.py.
